services/src/unit-test/java/uk/gov/moj/sdt/services/BulkSubmissionServiceTest.java

```diff
@@ -367,4 +456,108 @@ private void addValidIndividualRequest(final IBulkSubmission bulkSubmission, fin
+        ISdtBulkReferenceGenerator sdtBulkReferenceGeneratorMock = mock(SdtBulkReferenceGenerator.class);
+        ICacheable errorMessagesCache = Mockito.mock(ICacheable.class);
+
+
+        BulkSubmissionService bulkSubmission = new BulkSubmissionService(
+            genericDaoMock,bulkCustomerDaoMock, targetApplicationDaoMock,individualRequestsXmlParserMock,
+            messagingUtility,sdtBulkReferenceGeneratorMock,errorMessagesCache);
+
+
+        bulkSubmission.setErrorMessagesCache(errorMessagesCache);
+
+        Field errorMessagesCacheField = BulkSubmissionService.class.getDeclaredField("errorMessagesCache");
+        errorMessagesCacheField.setAccessible(true);
+        ICacheable result = (ICacheable) errorMessagesCacheField.get(bulkSubmission);
+
+        assertEquals(errorMessagesCache, result, "The errorMessagesCache should be set correctly");
+    }
+
+
+    @Test
+    public void setMockSdtBulkReferenceGeneratorTest() throws Exception {
+
+        bulkSubmissionService.setSdtBulkReferenceGenerator(mockSdtBulkReferenceGenerator);
+
+        Object result = this.getAccessibleField(BulkSubmissionService.class,"sdtBulkReferenceGenerator",
+                                                SdtBulkReferenceGenerator.class,bulkSubmissionService);
+
+        assertEquals(mockSdtBulkReferenceGenerator, result, "The sdtBulkReferenceGenerator should be set correctly");
+    }
+
+    @Test
+    public void setTargetApplicationDaoTest(){
+
+        bulkSubmissionService.setTargetApplicationDao(mockTargetApplicationDao);
+
+        ITargetApplicationDao result = bulkSubmissionService.getTargetApplicationDao();
+
+        assertEquals(mockTargetApplicationDao, result, "The TargetApplication should be set correctly");
+    }
+
+    @Test
+    public void setIndividualRequestXmlParserTest(){
+
+        IndividualRequestsXmlParser individualRequestsXmlParserMock = mock(IndividualRequestsXmlParser.class);
+        bulkSubmissionService.setIndividualRequestsXmlparser(individualRequestsXmlParserMock);
+
+        IndividualRequestsXmlParser result = bulkSubmissionService.getIndividualRequestsXmlParser();
+
+        assertEquals(individualRequestsXmlParserMock, result, "The TargetApplication should be set correctly");
+    }
+
+    @Test
+    public void setMessagingUtilityTest(){
+
+        bulkSubmissionService.setMessagingUtility(mockMessagingUtility);
+
+        IMessagingUtility result = bulkSubmissionService.getMessagingUtility();
+
+        assertEquals(mockMessagingUtility, result, "The MessageUtility should be set correctly");
+    }
+
+    @Test
+    public void setGenericDaoTest(){
+
+        bulkSubmissionService.setGenericDao(mockGenericDao);
+
+        IGenericDao result = bulkSubmissionService.getGenericDao();
+
+        assertEquals(mockGenericDao, result, "The GenericDao should be set correctly");
+    }
+
+    @Test
+    public void setBulkCustomerDaoTest(){
+
+        bulkSubmissionService.setBulkCustomerDao(mockBulkCustomerDao);
```

Comment:
The setUp() method already sets the bulk customer DAO to mockBulkCustomerDao, so setting it to the same thing doesn't necessarily prove that the set method has worked.  Create a new instance of the object and use that in the set method.

Reply:
done